sdk/messaging/azservicebus/admin/admin_client_rules.go

```diff
@@ -0,0 +1,633 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package admin
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal/atom"
+)
+
+// Rule specifies a message filter and action for a subscription.
+type Rule struct {
+	// Filter is the filter that will be used for Rule.
+	// Valid types: *SQLFilter, *CorrelationFilter, *FalseFilter, *TrueFilter
+	Filter RuleFilter
+
+	// Action is the action that will be used for Rule.
+	// Valid types: *SQLAction
+	Action RuleAction
+}
+
+// RuleFilter is a filter for a subscription rule.
+// Implemented by: *SQLFilter, *CorrelationFilter, *FalseFilter, *TrueFilter
+type RuleFilter interface {
+	ruleFilter()
+}
+
+// RuleAction is an action for a subscription rule.
+// Implemented by: *SQLAction
+type RuleAction interface {
+	ruleAction()
+}
+
+// SQLAction is an action that updates a message according to its
+// expression.
+type SQLAction struct {
+	// Expression is a SQL Expression
+	Expression string
+
+	// Parameters is a map of string to values of type string, number, or boolean.
+	Parameters map[string]interface{}
+}
+
+func (a *SQLAction) ruleAction() {}
+
+// UnknownRuleAction is an action type not yet handled by this SDK.
+// If you get this type back you should update to a newer version of the SDK
+// which properly represents this type.
+type UnknownRuleAction struct {
+	// Type is the Service Bus type for this action.
+	Type string
+
+	// RawXML is the raw XML for this action that could not be parsed.
+	RawXML string
+
+	// RawAttrs are attributes for the raw XML element that wasn't covered by our known properties.
+	RawAttrs []xml.Attr
```

Comment:
Why are the attributes separate, I would have thought them to be in `RawXML`.

Reply:
Agree, I'll unify.

Reply:
Unified. 

I also just made all the "RawXML" fields just be a byte slice, just to remove one unneeded conversion when calling xml.Unmarshal (should a user choose to do so).